Makefile

```diff
@@ -104,56 +107,61 @@ build-ccm-image: docker-pull-prerequisites ## Build controller-manager image.
 
-sub-build-node-image-windows-%:
+.PHONY: build-and-push-node-image-windows
+build-and-push-node-image-windows: ## Build node-manager image for Windows and push it to registry.
 	go build -a -o $(BIN_DIR)/azure-cloud-node-manager.exe ./cmd/cloud-node-manager
 	docker buildx build --pull --output=type=registry --platform windows/amd64 \
-		-t $(NODE_MANAGER_WINDOWS_IMAGE)-$* --build-arg OSVERSION=$* \
+		-t $(NODE_MANAGER_WINDOWS_IMAGE)-$(WINDOWS_OSVERSION) --build-arg OSVERSION=$(WINDOWS_OSVERSION) \
 		-f cloud-node-manager-windows.Dockerfile .
 
-.PHONY: build-and-push-all-windows-node-images
-build-and-push-all-windows-node-images: $(addprefix sub-build-node-image-windows-,$(ALL_WINDOWS_OSVERSIONS))
-
 .PHONY: build-ccm-e2e-test-image
 build-ccm-e2e-test-image: ## Build e2e test image.
 	docker build -t $(CCM_E2E_TEST_IMAGE) -f ./e2e.Dockerfile .
 
-.PHONY: build-images
-build-images: build-ccm-image build-node-image ## Build all images.
-
-.PHONY: image
-image: build-ccm-image build-node-image ## Build all images.
-
 .PHONY: push-ccm-image
-push-ccm-image: ## Push controller-manager image.
+push-ccm-image: build-ccm-image ## Push controller-manager image.
 	docker push $(IMAGE)
 
 .PHONY: push-node-image
 push-node-image: ## Push node-manager image for Linux.
-	docker push $(NODE_MANAGER_IMAGE)
+	docker push $(NODE_MANAGER_LINUX_FULL_IMAGE):$(IMAGE_TAG)-$(ARCH)
 
-.PHONY: push-node-image-windows
-push-node-image-windows: ## Push node-manager image for Windows.
-	docker push $(NODE_MANAGER_WINDOWS_IMAGE)
+.PHONY: release-ccm-e2e-test-image
+release-ccm-e2e-test-image: ## Build and release e2e test image.
+	docker build -t $(CCM_E2E_TEST_RELEASE_IMAGE) -f ./e2e.Dockerfile .
+	docker push $(CCM_E2E_TEST_RELEASE_IMAGE)
 
-.PHONY: push
-push: push-ccm-image push-node-image ## Push all images.
+hyperkube:  ## Build hyperkube image.
+ifneq ($(K8S_BRANCH), )
+	$(eval K8S_VERSION=$(shell REGISTRY=$(IMAGE_REGISTRY) BRANCH=$(K8S_BRANCH) hack/build-hyperkube.sh))
+	$(eval HYPERKUBE_IMAGE=$(IMAGE_REGISTRY)/hyperkube-amd64:$(K8S_VERSION))
+endif
+
+## --------------------------------------
+##@ All Arch or OS Version
+## --------------------------------------
+
+.PHONY: build-images
+build-images: build-ccm-image build-all-node-images ## Build all images.
+
+.PHONY: image
+image: build-ccm-image build-all-node-images ## Build all images.
 
 .PHONY: push-images
-push-images: push-ccm-image push-node-image ## Push all images.
+push-images: push-ccm-image push-all-node-images ## Push all images.
+
+.PHONY: push
+push: push-ccm-image push-all-node-images push-all-windows-node-images ## Push all images.
 
 .PHONY: push-node-manager-manifest
-push-node-manager-manifest: ## Create and push a manifest list containing all the Windows and Linux images.
+push-node-manager-manifest: push-all-node-images push-all-windows-node-images ## Create and push a manifest list containing all the Windows and Linux images.
```

Comment:
I did not manage to test push-node-manager-manifest as building windows image raised the following error
```
error: failed to solve: rpc error: code = Unknown desc = operating system is not supported
```

But test after removing windows related code works passed.

Reply:
Could you try adding the following lines to install qemu as part of the build?

```
docker run --privileged --rm tonistiigi/binfmt --uninstall qemu-aarch64
docker run --rm --privileged tonistiigi/binfmt --install all
```